xbmc/pvr/epg/EpgSearchFilter.cpp

```diff
@@ -114,7 +117,7 @@ bool CPVREpgSearchFilter::MatchDuration(const CPVREpgInfoTagPtr &tag) const
 
 bool CPVREpgSearchFilter::MatchStartAndEndTimes(const CPVREpgInfoTagPtr &tag) const
 {
-  return (tag->StartAsLocalTime() >= m_startDateTime && tag->EndAsLocalTime() <= m_endDateTime);
+  return (tag->StartAsLocalTime() >= GetStartDateTime() && tag->EndAsLocalTime() <= GetEndDateTime());
```

Comment:
I still don't like the `mutable` stuff but it's probably the easiest route for now. Ideally the start and end times should be set separately from the getters and setters, but at the right moment so that the values will be valid (unlike what happened in the constructor before).